StoreFront.sql

```diff
@@ -0,0 +1,119 @@
+CREATE DATABASE StoreFront;
+
+USE StoreFront;
+
+CREATE TABLE Categories(
+    CategoryId INT PRIMARY KEY NOT NULL AUTO_INCREMENT,
+    ParentId INT,
+    CategoryName VARCHAR(100)
+    );
+
+CREATE TABLE Product(
+    ProductId INT PRIMARY KEY NOT NULL AUTO_INCREMENT,
+    CategoryId INT,
+    Price INT,
+    Description VARCHAR(100),
+    Quantity INT,
+    Name VARCHAR(100),
+    FOREIGN KEY (CategoryId) REFERENCES Categories(CategoryId)
+    );
+
+CREATE TABLE ProductImages(
+    ProductId INT,
+    Image VARCHAR(100),
+    FOREIGN KEY (ProductId) REFERENCES Product(ProductId)
+    );
+
+CREATE TABLE Users(
+    UserId INT PRIMARY KEY NOT NULL AUTO_INCREMENT,
+    FirstName VARCHAR(100),
+    LastName VARCHAR(100),
+    EmailId VARCHAR(100) UNIQUE,
+    Password VARCHAR(100),
+    DateOfBirth DATE,
+    Gender VARCHAR(10) CHECK (Gender IN ("Male","Female"))
+    );
+
+CREATE TABLE UserContactDetails(
+    UserId INT,
+    ContactNumber VARCHAR(20),
+    FOREIGN KEY (UserId) REFERENCES Users(UserId)
+    );
+
+CREATE TABLE UserAddress(
+    UserId INT,

```

Comment:
This table should contain the mapping between Address Id and user Id. There should be a separate table for the address.